client/src/QAComponents/NewAnswer.jsx

```diff
@@ -22,64 +24,46 @@ const NewAnswer = (props) => {
   }
 
   const reportAnswer = (e) => {
-    const answerId = e.target.parentNode.parentNode.parentNode.id
+    const answerId = e.target.parentNode.parentNode.parentNode.parentNode.id
     e.preventDefault()
     axios.put('/qa/answers/report/?answer_id=' + answerId)
       .then(data => {
-        return alert('This Answer has been reported')
+        return setReport('Reported')
       })
       .catch(err => {
         console.error(err)
       })
   }
 
   const helpfulAnswer = (e) => {
-    const answerId = e.target.parentNode.parentNode.parentNode.id
+    const answerId = e.target.parentNode.parentNode.parentNode.parentNode.parentNode.id
     e.preventDefault()
-    axios.put('/qa/answers/helpful/?answer_id=' + answerId)
-      .then(data => {
-        // console.log(helpfulAns)
-        return setHelpfulness(helpfulness + 1)
-      })
-      .catch(err => {
-        console.error(err)
-      })
+    if (!helpful) {
+      setHelpful(true)
+      axios.put('/qa/answers/helpful/?answer_id=' + answerId)
+        .then(data => {
+          // console.log(helpfulAns)
+          return setHelpfulness(helpfulness + 1)
+        })
+        .catch(err => {
+          console.error(err)
+        })
+    } else {
+      alert('This answer has already been marked as "helpful"')
```

Comment:
I'm usually not a fan of alerts since they are notoriously obtrusive , but they are useful for bug testing.

Reply:
Thanks! This will be a high priority for optimization.